Introduce Zone Level

gcapalbo opened the pull request:
This introduces the Zone level into the location hierarchy and accounts for it where necessary in the facility-level SMS workflows. No change actually needed to be made to the order stockout process since it just looks at the district's parent location which will now be the zone.

For the warehouse runner, I took the approach of not storing (i.e., not aggregating) any data at the zone level since it's not a requirement.

Opening this for review, I still need to go through and make sure there's nowhere else that will fail with the introduction of a new level, and also add some tests.

@czue

czue commented on logistics_project/apps/malawi/loader.py:
```diff
@@ -137,7 +139,235 @@ def get_facility_export(file_handle):
+    If an error occurs, a FacilityLoaderValidationError is raised describing the
+    error and the row that had the error. If any exception is raised, no changes
+    are applied to the database.
+    """
```
wow nice doc string and error checking

czue commented on logistics_project/apps/malawi/util.py:
```diff
@@ -65,15 +65,22 @@ def hsas_below(location):
-                           Q(supply_point__supplied_by__location=location) | \
-                           Q(supply_point__supplied_by__supplied_by__location=location) | \
-                           Q(supply_point__supplied_by__supplied_by__supplied_by__location=location))
+        if location.type_id == config.LocationCodes.HSA:
```
thanks this seems much more explicit and probably more performant.

gcapalbo commented:
@czue i went through and updated all remaining references that i could find that had assumptions about the location hierarchy. also did some local testing and added some automated tests. this should be good to go now from my side

czue commented on logistics_project/apps/malawi/tests/util_functions.py:
```diff
@@ -0,0 +1,64 @@
+        result = hsas_below(Location.objects.get(code=location_code))
+        self.assertEqual(list(result), expected_list)
+
+    def test_hsas_below(self):
```
nice, thanks for adding these

czue commented:
thanks!
